src/index.ts

```diff
@@ -34,6 +36,26 @@ export async function resolve(
 ): Promise<ResolveResponse> {
   const { parentURL = baseURL } = context;
 
+  let forceRelative = false;
+  if (TSConfig && TSConfig.paths) {
+    for (const tsPath of Object.keys(TSConfig.paths)) {
+      const tsPathKey = tsPath.replace('/*', '');
+      if (specifier.startsWith(tsPathKey)) {
+        const pathSpecifier = TSConfig.paths[tsPath][0].replace(
+          '/*',
+          specifier.split(tsPathKey)[1],
+        );
+
+        forceRelative = true;
+        specifier = resolvePath(
+          rootModulePath,
+          TSConfig.baseUrl!,
+          pathSpecifier,
+        );
+      }
+    }
+  }
+
```

Comment:
It's a bit of a hack, but I need to swap out the "/*" in the key and the path, and then force relative so that the fileFinder function runs since most people aren't using extension ESModule imports in Node.JS. And I need to resolve the path from the TSConfig path because of how the baseUrl and stuff is used in TypeScript. If I was the person writing the TypeScript side of the Paths feature I would have done it diferently.